pkg/api/types.go

```diff
@@ -1004,6 +1008,30 @@ type Service struct {
 	Status ServiceStatus `json:"status,omitempty"`
 }
 
+// ServiceAccount binds together:
+// * a name, understood by users, and perhaps by peripheral systems, for an identity
+// * a principal that can be authenticated and authorized
+// * a security context, which defines the Linux Capabilities, User IDs, Groups IDs, and other capabilities and controls on interaction with the file system and OS.
+// * a set of secrets
+type ServiceAccount struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// APISecret is the name of the secret to use to identify this ServiceAccount to the API
+	// It must be of type SecretTypeServiceAccountToken
+	APISecret string `json:"apiSecret"`
+	// Secrets is the list of secrets allowed to be used by pods running using this ServiceAccount
+	Secrets []string `json:"secrets"`
```

Comment:
The `apiSecret` field did feel messy... open to suggestions. My thinking in making it a special field:

1. It is sort of special... the secret it refers to has to be of a certain type, and two different service accounts shouldn't reference the same ServiceAccountToken secret.
2. Explicitly naming the secret that holds the ServiceAccount's token lets the admission controller that sets up the VolumeMount do so based only on the info in the ServiceAccount, without also listing/searching secrets.
3. Avoids ambiguity if there are more than one ServiceAccountToken secret for a ServiceAccount (which is intentionally possible)

As to the name, `serviceAccountTokenSecret` felt too long (though that might be better than `apiToken` now that I think about it), and `tokenSecret` not descriptive enough.